manim/mobject/types/vectorized_mobject.py

```diff
@@ -1364,9 +1364,12 @@ def get_arc_length(self, sample_points_per_curve: Optional[int] = None) -> float
 
     # Alignment
     def align_points(self, vmobject):
-        # This probably makes the current vmobject and the given one have the same number of points,
-        # by adding extra points to the last sub-path. This method is never used in the whole library.
+        """Adds points to self and vmobject so that they both have the same number of subpaths, each of the matching lengths.
+        Points are added either by subdividing curves evenly along the subpath, or by creating new subpaths consisting
+        of a single point repeated.
+        """
         self.align_rgbas(vmobject)
+        # TODO: This shortcut can be a bit over eager. What if they have the same length, but different subpath lengths?
         if self.get_num_points() == vmobject.get_num_points():
             return
```

Comment:
I'm pretty sure this is buggy, but it's causing a test regression and it's not really related to the rest of this PR, so I'm not touching it.